### PR TITLE
feat: carma outlier detection method

### PR DESCRIPTION
**Pull request by d0choa**
CARMA outlier detection method:

- [x] Migration of the method from @addramir [repo](https://github.com/addramir/carmapy_light)
- [x] Documentation page
- [x] Testing

### CODE REVIEW

**Comment by d0choa** [comment]
You can easily get 100% testing coverage by adding additional examples that evaluate the conditions (if statements) that you are currently missing.

Have a look at the red lines here:
https://app.codecov.io/gh/opentargets/genetics_etl_python/pull/281/blob/src/otg/method/carma.py

**Comment by addramir** [comment]
@d0choa I have added two test. For now this is the biggest coverage I can get.

**Comment by d0choa** [comment]
@DSuveges we need approval here

File: src/otg/method/carma.py [hunk 1]
```diff
@@ -0,0 +1,883 @@
...
+            outlier_BF_index (float): Bayes Factor threshold for identifying outliers.
+
+        Returns:
+            dict[str, Any]: A dictionary containing the following results:
+                - PIPs: A numeric vector of posterior inclusion probabilities (PIPs) for all SNPs.
+                - B_list: A dataframe containing the marginal likelihoods and the corresponding model space.
+                - Outliers: A list of outlier SNPs.
+        """
+        # t0 = pd.Timestamp.now()
+
```
Comment (tskir) [inline src/otg/method/carma.py:53]: Looks like a forgotten debug output (the same for the next few suggestions)
```suggestion
```

File: src/otg/method/carma.py [hunk 2]
```diff
@@ -0,0 +1,883 @@
...
+        p_snp = len(z)
+        epsilon_list = epsilon_threshold * p_snp
+        all_epsilon_threshold = epsilon_threshold * p_snp
+
+        # print("ordering pizza...")
+        # print(f"N SNPs is {p_snp}")
```
Comment (tskir) [inline src/otg/method/carma.py:59]:
```suggestion
```

File: src/otg/method/carma.py [hunk 3]
```diff
@@ -0,0 +1,883 @@
...
+            tau=tau,
+            num_causal=num_causal,
+            inner_all_iter=all_inner_iter,
+            outlier_BF_index=outlier_BF_index,
+        )
+
+        # t1 = pd.Timestamp.now() - t0
+        # print(f"Zero step is finished in {round(t1.total_seconds(), 2)} sec")
+        # print(f"Expecting to finish in {round(t1.total_seconds()*all_iter, 2)} sec")
+
```
Comment (tskir) [inline src/otg/method/carma.py:77]:
```suggestion
```

File: src/otg/method/carma.py [hunk 4]
```diff
@@ -0,0 +1,883 @@
...
+        results_list = {
+            "PIPs": pip,
+            "B_list": all_C_list["B_list"],
+            "Outliers": all_C_list["conditional_S_list"],
+        }
+
+        # print("pizza ordered!")
+        # t1 = pd.Timestamp.now() - t0
+
+        # print(f"FM time is {round(t1.total_seconds(), 2)} sec")
```
Comment (tskir) [inline src/otg/method/carma.py:119]:
```suggestion
```

**Review by tskir — APPROVED** [review]
I added some minor style comments, but overall everything looks good to me

**Comment by addramir** [comment]
@DSuveges could you please help to resolve the conflict in poetry.lock?

**Pull request merged**